kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/spec/resolveConfig.kt

```diff
@@ -72,7 +72,11 @@ internal fun resolveConfig(
       threads = threads,
       invocations = invocations,
       timeout = timeout,
-      invocationTimeout = invocationTimeout,
+      // note: the invocation timeout cannot be larger than the test case timeout
+      invocationTimeout = when {
+         timeout != null && invocationTimeout != null -> invocationTimeout.coerceAtMost(timeout)
```

Comment:
Should we just error here so the user is clear it's setup wrong ?

Reply:
Done! But I'm not sure if I throw a correct exception (`IllegalStateException`) and if I do it in the proper place.